sw/device/examples/aes_serial/aes_serial.c

```diff
@@ -0,0 +1,271 @@
+  }
+  return SIMPLE_SERIAL_OK;
+}
+
+/**
+ * Convert `num` bytes `from` hex `to` binary format.
+ *
+ * `from` size is expected to by twice as big as `to`.
+ *
+ * @param from input buffer for hex formatted characters.
+ * @param to   output binary buffer.
+ * @param num  number of characters in output buffer.
+ *
+ * @return SIMPLE_SERIAL_OK on success, SIMPLE_SERIAL_ERROR otherwise.
+ */
+static uint8_t a2b_hex(const char *from, char *to, size_t num) {
+  for (int i = 0; i < num; ++i) {
+    char hi, lo;
+    if (hex_value(from[i * 2], &hi) || hex_value(from[i * 2 + 1], &lo)) {
+      return SIMPLE_SERIAL_ERROR;
+    }
+    to[i] = ((hi & 0xFF) << 4) | (lo & 0xFF);
+  }
+  return SIMPLE_SERIAL_OK;
+}
+
+/**
+ * Send `data` to UART in hex byte format.
+ */
+static void print_hex_buffer(const char *data, size_t data_len) {
+  static const char b2a_hex_values[16] = {'0', '1', '2', '3', '4', '5',
+                                          '6', '7', '8', '9', 'A', 'B',
+                                          'C', 'D', 'E', 'F'};
+  for (int i = 0; i < data_len; ++i) {
+    uart_send_char(b2a_hex_values[data[i] >> 4]);
+    uart_send_char(b2a_hex_values[data[i] & 0xF]);
+  }
+}
+
+/**
+ * Send simple serial command response via UART.
+ *
+ * @param cmd_tag  Simple serial command tag.
+ * @param data     Response data. Converted to hex format by this function.
+ * @param data_len data length.
+ */
+static void print_cmd_response(const char cmd_tag, const char *data,
+                               size_t data_len) {
+  uart_send_char(cmd_tag);
+  print_hex_buffer(data, data_len);
+  uart_send_char('\n');
+}
+
+/**
+ * Return number of AES blocks from `plain_text_len`.
+ */
+static size_t get_num_blocks(size_t plain_text_len) {
+  size_t num_blocks = plain_text_len / AES_BLOCK_SIZE;
+  if ((plain_text_len - (num_blocks * AES_BLOCK_SIZE)) > 0) {
+    ++num_blocks;
+  }
+  return num_blocks;
+}
+
+/**
+ * Let the AES unit process multiple blocks.
+ *
+ * @param data_out   pointer to output buffer.
+ * @param data_in    pointer to input buffer.
+ * @param num_blocks number of 16-byte blocks to process.
+ */
+static void aes_process_blocks(void *data_out, const void *data_in,
+                               size_t num_blocks) {
+  // TODO: Enable block parallel processing.
+  for (int i = 0; i < num_blocks; ++i) {
+    aes_data_put_wait(data_in + i * AES_BLOCK_SIZE);
+    aes_data_get_wait(data_out + i * AES_BLOCK_SIZE);
+  }
+}
+
+/**
+ * Simple serial set AES key command.
+ *
+ * @param aes_cfg AES configuration object.
+ * @param key     AES key.
+ * @param key_len AES key length.
+ *
+ * @return SIMPLE_SERIAL_OK on success, SIMPLE_SERIAL_ERROR otherwise.
+ */
+static uint8_t simple_serial_set_key(const aes_cfg_t *aes_cfg, const char *key,
+                                     size_t key_len) {
+  if (key_len != AES_KEY_SIZE) {
+    return SIMPLE_SERIAL_ERROR;
+  }
+  aes_clear();
+  aes_key_put((const void *)key, aes_cfg->key_len);
```

Comment:
I would put an `aes_idle()` between `aes_clear()` and `aes_key_put()` just to be on the safe side.